.cargo/config.toml

```diff
@@ -27,3 +27,6 @@ rustflags = ["--cfg", "espidf_time64"]
 
 [unstable]
 build-std = ["std", "panic_abort"]
+
+[env]
+ESP_IDF_VERSION = "v5.4.1"
```

Comment:
This is how the [official rust esp-idf template](https://github.com/esp-rs/esp-idf-template/blob/master/cargo/.cargo/config.toml#L14-L21) sets the version. At build time, this will be visible to everyone (build.rs, embuild, esp-idf build system, etc)